pkg/controller/network_policy.go

```diff
@@ -558,13 +647,74 @@ func (c *Controller) fetchPolicySelectedAddresses(namespace, protocol string, np
 			for _, podIP := range pod.Status.PodIPs {
 				if podIP.IP != "" && util.CheckProtocol(podIP.IP) == protocol {
 					selectedAddresses = append(selectedAddresses, podIP.IP)
+
+					svcs, err := c.servicesLister.Services(pod.Namespace).List(labels.Everything())
+					if err != nil {
+						klog.Errorf("failed to list svc, %v", err)
+						return nil, nil, fmt.Errorf("failed to list svc, %v", err)
+					}
+					if len(svcs) == 0 {
+						continue
+					}
+					klog.Infof("svc is %v", svcs)
+					svcIPs, err := svcMatchPods(svcs, pod, protocol)
+					if err != nil {
+						return nil, nil, err
+					}
+					klog.Infof("svcIPs is %v", svcIPs)
+					if len(svcIPs) == 0 {
+						continue
+					}
+					selectedAddresses = append(selectedAddresses, svcIPs...)
 				}
 			}
 		}
 	}
+
 	return selectedAddresses, nil, nil
 }
 
+func svcMatchPods(svcs []*corev1.Service, pod *corev1.Pod, protocol string) ([]string, error) {
+	matchSvcs := []string{}
+	// find  svc ip by pod's info
+	for _, svc := range svcs {
+		isMatch, err := isSvcMatchPod(svc, pod)
+		if err != nil {
+			return nil, err
+		}
+		if isMatch {
+			protocolClusterIPs := getProtocolSvcIp(svc.Spec.ClusterIPs, protocol)
+			if len(protocolClusterIPs) != 0 {
+				matchSvcs = append(matchSvcs, protocolClusterIPs...)
+			}
+		}
+	}
+	return matchSvcs, nil
+}
+func getProtocolSvcIp(clusterIPs []string, protocol string) []string {
+	protocolClusterIPs := []string{}
+	for _, clusterIP := range clusterIPs {
+		if clusterIP != "" && util.CheckProtocol(clusterIP) == protocol {
+			protocolClusterIPs = append(protocolClusterIPs, clusterIP)
+		}
```

Comment:
We should check whether the `clusterIP` is `None`.

Reply:
ok